### PR TITLE
fix: fix decimal comparison and aggregation

### PR DESCRIPTION
**Pull request by iajoiner**
# Rationale for this change
It is necessary for us to be able to do comparison and aggregation on decimal columns.

# What changes are included in this PR?
- add `signed_cmp` to `Scalar` definition to do signed Scalar comparison (i.e. `S::MAX_SIGNED` is the largest number and everything "beyond" it wraps around into negative numbers)
- allow decimal comparison
- allow `SUM` of decimals

# Are these changes tested?
Yes.

### CODE REVIEW

File: crates/proof-of-sql/src/base/scalar/mod.rs
Comment (tlovell-sxt) [inline crates/proof-of-sql/src/base/scalar/mod.rs:24]: I'm trying to think if a new trait is the right way to go here, or if it's a new type. For example

```rust
pub struct SignedScalar<S>(S) where S: Scalar;

// implement various num_traits and operations for `SignedScalar`, including `Ord`

// use `SignedScalar` in Decimal75 columns instead of plain Scalar
```

I think something like that is a bigger lift, but I think it has a couple benefits. Namely,
1. The new type makes it very clear what the meaning of the underlying data is, whereas a new trait is sort of agnostic about this (it could be either)
2. Allows for using the `SignedScalar` in generic code w/ trait bounds like `T: Ord` without unexpected results (the type is very clear about which order implementation is the right one)

The drawbacks would be, again, it's a bigger lift. Another one I can think of is that it might be hard to understand which one is the correct one to use elsewhere in the code base, or as we develop new code. I think for this purpose it would be good if `SignedScalar` didn't actually implement `Scalar`, so that we don't make the mistake of using it in the proving protocols, but it could implement `Into<S> where S: Scalar`. I don't know off the top of my head if this thought is compatible with all of our current code.

Comment (iajoiner) [reply crates/proof-of-sql/src/base/scalar/mod.rs:24]: I think the essence of the problem is that any `Scalar` just like bytes and vectors of bytes have both signed and unsigned forms of comparison.

File: crates/proof-of-sql/src/sql/ast/group_by_util.rs
Comment (tlovell-sxt) [inline crates/proof-of-sql/src/sql/ast/group_by_util.rs:119]: Could you actually lay out the cases for this? Just so when we add new types it'll cause a compiler error until we update this code?

```suggestion
        Column::VarChar(_) | Column::TimestampTZ(_) | Column::Boolean(_) => unreachable!("SUM can not be applied to non-numeric types"),
```

**Review by tlovell-sxt — COMMENTED** [review]
Just one thing from me

File: crates/proof-of-sql/src/sql/ast/group_by_util.rs
Comment (iajoiner) [reply crates/proof-of-sql/src/sql/ast/group_by_util.rs:119]: Sure!

**Review by tlovell-sxt — APPROVED** [review]
LGTM, idk if you want an additional review from Jay or anything

**Review by tlovell-sxt — APPROVED** [review]

**Pull request merged**
